astropy/io/fits/convenience.py

```diff
@@ -709,6 +710,98 @@ def info(filename, output=None, **kwargs):
     return ret
 
 
+def diff(inputa, inputb, *args, **kwargs):
+    """
+    Compare two parts of a fits file, including entire FITS files, and
+    FITS HDU objects.
+
+    Parameters
+    ----------
+    inputa : str of `HDUList`
+        The filename of a FITS file on disk, or an `HDULIST` object.
+
+    inputb : str or `HDUList`
+        The filename of a FITS file on disk, or an `HDULIST` object.
+
+    ext
+        The rest of the arguments are for extension specification if your
+        inputA and inputB are string filenames (will not work if inputA and
+        inputB are HDU objects. They are flexible and are best illustrated
+        by examples.
+
+        By extension number::
+            diff('inA.fits', 'inB.fits', 0)      # the primary HDU
```

Comment:
If I remember correctly, you need a blank line between `::` and the block quote.